### PR TITLE
WIP: BDF Fixes

### PR DESCRIPTION
**Pull request by utkarsh530**
Hi, I am trying to fix random errors in BDF popping in travis build. I removed some `similar` allocations which I found to be suspicious. Testing more fixes.
@ChrisRackauckas @kanav99

### CODE REVIEW

File: src/caches/bdf_caches.jl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@cache mutable struct ABDF2ConstantCache{N,dtType,rate_prototype} <: OrdinaryDiffEqConstantCache
+  @cache mutable struct ABDF2ConstantCache{N,dtType,rate_prototype} <: OrdinaryDiffEqConstantCache
```
Comment (ChrisRackauckas) [inline src/caches/bdf_caches.jl]: intentional?

**Comment by ChrisRackauckas** [comment]
Those are definitely suspicious! Good catch.

File: src/caches/bdf_caches.jl (hunk 1 above)
Comment (ChrisRackauckas) [inline src/caches/bdf_caches.jl]:
```suggestion
@cache mutable struct ABDF2ConstantCache{N,dtType,rate_prototype} <: OrdinaryDiffEqConstantCache
```

**Comment by ChrisRackauckas** [comment]
It's at least an improvement, so I'm merging.

**Comment by utkarsh530** [comment]
Thanks! 😃

**Pull request merged**